support/releaseinfo/registryclient/client.go

```diff
@@ -223,36 +232,107 @@ func ExtractImageFilesToDir(ctx context.Context, imageRef string, pullSecret []b
 	rt, err := rest.TransportFor(&rest.Config{})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create secure transport: %w", err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create secure transport: %w", err)
 	}
 	insecureRT, err := rest.TransportFor(&rest.Config{TLSClientConfig: rest.TLSClientConfig{Insecure: true}})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create insecure transport: %w", err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create insecure transport: %w", err)
 	}
 	credStore, err := dockercredentials.NewFromBytes(pullSecret)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse docker credentials: %w", err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to parse docker credentials: %w", err)
 	}
 	registryContext := registryclient.NewContext(rt, insecureRT).WithCredentials(credStore).
 		WithRequestModifiers(transport.NewHeaderRequestModifier(http.Header{http.CanonicalHeaderKey("User-Agent"): []string{rest.DefaultKubernetesUserAgent()}}))
 
 	ref, err := reference.Parse(imageRef)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
 	}
 	repo, err := registryContext.Repository(ctx, ref.DockerClientDefaults().RegistryURL(), ref.RepositoryName(), false)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create repository client for %s: %w", ref.DockerClientDefaults().RegistryURL(), err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create repository client for %s: %w", ref.DockerClientDefaults().RegistryURL(), err)
 	}
-	firstManifest, location, err := manifest.FirstManifest(ctx, ref, repo)
+	return repo, ref, nil
+}
+
+func GetManifest(ctx context.Context, imageRef string, pullSecret []byte)(distribution.Manifest, error) {
+	repo, ref, err := GetRepoSetup(ctx, imageRef, pullSecret)
+
+	var srcDigest digest.Digest
+	if len(ref.ID) > 0 {
+		srcDigest = digest.Digest(ref.ID)
+	} else if len(ref.Tag) > 0 {
+		desc, err := repo.Tags(ctx).Get(ctx, ref.Tag)
+		if err != nil {
+			return nil, err
+		}
+		srcDigest = desc.Digest
+	} else {
+		return nil, err
+	}
+	manifests, err := repo.Manifests(ctx)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to obtain root manifest for %s: %w", imageRef, err)
+		return nil, err
 	}
-	_, layers, err := manifest.ManifestToImageConfig(ctx, firstManifest, repo.Blobs(ctx), location)
+	digestsManifest, err := manifests.Get(ctx, srcDigest, manifest.PreferManifestList)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to obtain image layers for %s: %w", imageRef, err)
+		return nil, err
+	}
+
+	return digestsManifest, nil
+}
+
+func IsMultiArchManifestList(ctx context.Context, imageRef string, pullSecret []byte) (bool, error) {
```

Comment:
nit: please document funcs specially public ones.

Reply:
we need to lint this in CI, but please start comment with function name by convention, i.e. IsMultiArchManifestList Determines...

Reply:
Ok I will try to remember to start doing that from now on. I'll update the function comments here as well.